### PR TITLE
fix(firmware): don't treat unparseable version as "too old" (#3726)

### LINKED ISSUE #3726: [Bug]: Firmware too old popup
### Checklist

- [x] I am able to reproduce the bug with the latest version.

- [x] I made sure that there are no existing **OPEN or CLOSED issues** which I could contribute my information to.

- [x] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.

- [x] This issue contains only one bug.

- [x] I have read and understood the **Contribution Guidelines**.

- [x] I agree to follow this project's Code of Conduct

### Affected app version

2.7.6+

### Affected node firmware version

2.7.12? 

### Steps to reproduce the bug

Seems to be a race condition, but I saw it again several times yesterday on HEAD. 
We should try and track down what causes this, given that it locks you out fully for a session despite being wrong.

### PR DESCRIPTION
**Pull request by jamesarich**
## Summary
Fixes #3726 — the false "firmware too old" popup that force-disconnects devices running perfectly valid firmware.

`DeviceVersion.parseVersion()` returns `0` as a sentinel for an **unparseable** string, but `FirmwareVersionCheck` compared that `0` directly against the absolute-minimum version. A transient or garbled firmware string arriving mid-handshake therefore parsed to `0`, satisfied `0 < ABS_MIN`, and fired the **blocking** alert whose `onConfirm` disconnects the device — matching reports on hardware running e.g. 2.6.11.

## Change
- Extract the decision into a pure, unit-testable `DeviceVersion.checkStatus` returning `OK / SHOULD_UPDATE / TOO_OLD / UNKNOWN`.
- An unparseable version (int `0`) is now `UNKNOWN` — logged and skipped, never a forced disconnect.
- `FirmwareVersionCheck` consumes the status instead of raw comparisons.

## Testing
- New regression tests in `DeviceVersionTest` (unparseable → `UNKNOWN`, not `TOO_OLD`; known-version classification).
- `:core:model:allTests` (161 tests) green; `:core:ui` compiles; detekt + spotless clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added firmware version check status classification (OK, SHOULD_UPDATE, TOO_OLD, UNKNOWN) and exposed it for consistent use across the app.
* **Bug Fixes**
  * Prevented unparseable or transient firmware versions from being treated as “too old.”
  * Updated firmware alerts so blocking or update-required messages show only for actionable statuses; unknown statuses now log only.
* **Tests**
  * Added regression coverage for known and unparseable firmware version inputs, including boundary cases.

### CODE REVIEW

File: core/ui/src/commonMain/kotlin/org/meshtastic/core/ui/component/FirmwareVersionCheck.kt [hunk 1]
```diff
@@ -62,33 +63,44 @@ fun FirmwareVersionCheck(viewModel: UIViewModel) {
...
-                                    "device: $curVer < min: ${DeviceVersion(DeviceVersion.MIN_FW_VERSION)}"
+
+                            FirmwareCheckStatus.SHOULD_UPDATE -> {
+                                Logger.w {
+                                    "[FW_CHECK] Firmware should update - " +
+                                        "device: $curVer < min: ${DeviceVersion(DeviceVersion.MIN_FW_VERSION)}"
+                                }
+                                val title = getString(Res.string.should_update_firmware)
+                                val message = getString(Res.string.should_update, latestStableFirmwareRelease.asString)
+                                viewModel.showAlert(title = title, message = message, onConfirm = {})
```
[the remark this answers is not in the document]
Comment (jamesarich) [reply core/ui/src/commonMain/kotlin/org/meshtastic/core/ui/component/FirmwareVersionCheck.kt]: Not applying: this file imports `org.jetbrains.compose.resources.getString` directly (line 24) — the official Compose Multiplatform library's suspend function — not this repo's own blocking `org.meshtastic.core.resources.getString` wrapper (which wraps `runBlocking` and is the one the coding guideline actually targets). The one imported here is already non-blocking; `feature/docs/.../DocsNavigation.kt` uses the identical import + `LaunchedEffect` pattern as existing precedent. Also note this code predates this PR's diff — it isn't something introduced here.

File: core/ui/src/commonMain/kotlin/org/meshtastic/core/ui/component/FirmwareVersionCheck.kt [hunk 2]
```diff
@@ -62,33 +63,44 @@ fun FirmwareVersionCheck(viewModel: UIViewModel) {
...
-                            val message = getString(Res.string.should_update, latestStableFirmwareRelease.asString)
-                            viewModel.showAlert(title = title, message = message, onConfirm = {})
-                        } else {
-                            Logger.i { "[FW_CHECK] Firmware version OK - device: $curVer meets requirements" }
+
+                            // An unparseable/transient version must not disconnect the device (#3726); just log it.
+                            FirmwareCheckStatus.UNKNOWN ->
+                                Logger.w {
+                                    "[FW_CHECK] Firmware version unparseable - device raw: $fwVersion, skipping"
+                                }
```
[the remark this answers is not in the document]
Comment (jamesarich) [reply core/ui/src/commonMain/kotlin/org/meshtastic/core/ui/component/FirmwareVersionCheck.kt]: Fixed in 064d69c55 — removed the `isNotBlank()` filter so blank handshake values now reach the `FirmwareCheckStatus.UNKNOWN` branch and get logged (`DeviceVersion("").checkStatus` was already tested as `UNKNOWN`).
